### DISCUSSION TITLE
Add menu option to remove trailing spaces

### DISCUSSION THREAD

**Discussion by gnijs** [tsujan/FeatherPad#798] [Ideas]
I understand there is already an option to remove trailing spaces when saving a file.
However, I prefer to remove trailing spaces when editing.
Often when copy pasting some text from a browser, there are trailing spaces.
To be able to work on a clean file, it would be nice if I can use a function to remove those trailing spaces, without the need of saving and re-opening the file.

Either a menu or context menu option to remove trailing spaces would be nice :-)

**Comment by tsujan** [comment]
How trailing spaces are removed depends on the programming language, which is determined only after saving.

**Comment by tsujan** [comment]
BTW, you could easily remove trailing spaces of an unsaved document by toggling Regular Expression button on the search-bar (Rx) and replacing `\s+\z` with an empty string in the Replacement dock everywhere (`F10`).

**Reply by gnijs** [reply]
Thanks for the feedback @tsujan . I'll remove trailing spaces like you mentioned.
